### PR TITLE
`mean_relative_error` supports complex label

### LINKED ISSUE #14658: tf.metrics.mean_relative error doesn't handle complex inputs
The offending line is here: 

https://github.com/tensorflow/tensorflow/blob/9c4bdb865452e418a1d69cd5f5cdccb51d6a0e1d/tensorflow/python/ops/metrics_impl.py#L1057

`array_ops.where` expects the second and third inputs to be of the same shape and **type**. Unfortunately, when the `labels` input is complex, `array_ops.zeros_like(labels)` returns an (unnecessarily) complex tensor, whereas the third input returns a float tensor. 

I haven't checked, but suspect similar problems might occur with other metrics (and perhaps elsewhere).

### PR DESCRIPTION
**Pull request by facaiy**
Fix #14658 

### How to test

+ [x] add test case
+ [ ] pass all tests

### CODE REVIEW

File: tensorflow/python/ops/metrics_impl.py [hunk 1]
```diff
@@ -174,8 +174,8 @@ def _maybe_expand_labels(labels, predictions):
         lambda: labels)
 
 
-def _safe_div(numerator, denominator, name):
-  """Divides two values, returning 0 if the denominator is <= 0.
+def _safe_div(numerator, denominator, name=None):
+  """Divides two Tensors, returning 0 if the denominator is <= 0.
```
Comment (drpngx) [inline tensorflow/python/ops/metrics_impl.py]: please specify that this is component-wise division (aka Schur or Hadamard).

**Review by drpngx — CHANGES_REQUESTED** [review]

**Comment by drpngx** [comment]
I'd rather we didn't do this. Performing division by zero raises the FP exception flag and makes for a very slow operation. Either we have a new safe operator in C++ or we should ask user-level awareness.

**Comment by facaiy** [comment]
Yes, I found `_safe_div` exists in many modules, for example, `losses`, `metrics` and so on. Creating a C++ op perhaps is a better solution, at least for me.

**Comment by facaiy** [comment]
It seems that `safe_div` has been implemented in C++ side, however it only supports integer now.  

https://github.com/tensorflow/tensorflow/blob/ac7387b48674fdeb11e7a4fa595c88b5e2afe16e/tensorflow/core/kernels/cwise_ops.h#L702-L705

https://github.com/tensorflow/tensorflow/blob/ac7387b48674fdeb11e7a4fa595c88b5e2afe16e/tensorflow/core/kernels/cwise_ops.h#L119-L137

**Comment by drpngx** [comment]
OK, then should we implement the float version in `C++` then?

**Comment by facaiy** [comment]
Yes, I think so. I'll open a issue later, and let's discuss the details there. Thank you for your help, @drpngx .

**Comment by drpngx** [comment]
Thanks!

**Pull request closed without merging**